source/dub/project.d

```diff
@@ -563,7 +563,8 @@ class Project {
 					(VersionRange range) {
 						// See `dub.recipe.selection : SelectedDependency.fromYAML`
 						assert(range.isExactVersion());
-						return m_packageManager.getPackage(dep.name, vspec.version_);
+						auto tmp = m_packageManager.getPackage(basename, vspec.version_);
+						return resolveSubPackage(tmp, subname, true);
```

Comment:
This fixes the testcase in #2973. Not sure if this is the right fix though; `PackageManager.getPackage()` returning the parent package when given a subpackage name doesn't make a lot of sense IMO.

Reply:
Yeah it doesn't make a lot of sense but it's always been this way. It was one of the things I wanted to fix when I introduced `PackageName` but never got around to it.
For dependency resolution we always store references to parent package, not subpackage, because all subpackages should resolve to the same parent. I would need to inspect the code to see if that is really the right fix.

Reply:
Okay fine if it's always been like that.